blockchain/common/common.go

```diff
@@ -49,17 +49,23 @@ type FluxAggregatorState struct {
 	CanSubmit     bool
 }
 
-type RunlogRequest struct {
-	Params           map[string]string
-	Payment          string
-	RequestId        string
-	CallbackFunction string
-}
+type RunlogRequest map[string]interface{}
 
 type Manager interface {
-	Request(t string) (response interface{}, err error)
-	Subscribe(ctx context.Context, t string, ch chan<- interface{}) (err error)
-	CreateJobRun(t string, v interface{}) (map[string]interface{}, error)
+	Stop()
+}
+
+type FluxMonitorManager interface {
+	Manager
+	GetState(ctx context.Context) (*FluxAggregatorState, error)
+	SubscribeEvents(ctx context.Context, ch chan<- interface{}) error
+	CreateJobRun(roundId uint32) map[string]interface{}
+}
+
+type RunlogManager interface {
+	Manager
+	SubscribeEvents(ctx context.Context, ch chan<- RunlogRequest) error
+	CreateJobRun(request RunlogRequest) map[string]interface{}
```

Comment:
Instead of having a very generic interface with a lot of `interface{}` types, I've split runlog and FM into two different interfaces, so they can be implemented more precisely and gives us more type safety.